apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -58,13 +58,35 @@ type CloudService interface {
 	Cloud(ctx context.Context, name string) (*cloud.Cloud, error)
 }
 
+// ModelService provides access to the model.
+type ModelService interface {
+	// WatchModelCloudCredential returns a new NotifyWatcher watching for changes that
+	// result in the cloud spec for a model changing. The changes watched for are:
+	// - updates to model cloud.
+	// - updates to model credential.
+	// - changes to the credential set on a model.
+	// The following errors can be expected:
+	// - [modelerrors.NotFound] when the model is not found.
+	WatchModelCloudCredential(ctx context.Context, modelUUID model.UUID) (corewatcher.NotifyWatcher, error)
+}
+
+// ModelInfoService provides access to the model info.
+type ModelInfoService interface {
+	// GetModelInfo returns the readonly model information for the model in
+	// question.
+	GetModelInfo(ctx context.Context) (model.ModelInfo, error)
+}
+
 type CredentialValidatorAPI struct {
 	*credentialcommon.CredentialManagerAPI
 
 	logger            logger.Logger
-	backend           StateAccessor
+	backend           credentialcommon.StateBackend
 	cloudService      CloudService
 	credentialService CredentialService
+	modelService      ModelService
+	modelInfoService  ModelInfoService
+	modelUUID         model.UUID
```

Comment:
We don't need the modelUUID.
One use case will go away when we get rid of credential tag.
The other user case will go away when a credential watcher closure is used.

Reply:
Hi Ian, just wondering is there a reason why we are avoiding the passing in of model uuid?

Reply:
It's not needed.
We know at the time we register the facade what the model should be. We don't want to open a possible security hole by allowing the facade to use a different model uuid to call the domain services. So using the pattern as described in the comments ensures the facade is given what it needs to only operate on the correct model.

Reply:
I see, got it, thanks for the clarifications Ian! :)